incident_viewer_demo/src/LineChart/LineChart.jsx

```diff
@@ -647,35 +710,60 @@ class LineChart extends PureComponent {
     this.badIncidentLine.selectAll('*').remove();
 
     this.updateIncidentArrow();
+    this.updateIncidentShading();
 
-    const goodBinding = this.goodIncidentLine.selectAll('g').data(goodIncidentSeriesArray);
-    const badBinding = this.badIncidentLine.selectAll('g').data(badIncidentSeriesArray);
-
-    // ENTER
-    const goodEntering = goodBinding.enter()
-      .append('path')
-        .attr('d', incidentLineGenerator(goodIncidentSeriesArray));
-    const badEntering = badBinding.enter()
-      .append('path')
-        .attr('d', incidentLineGenerator(badIncidentSeriesArray));
+    // LINES
+    this.goodIncidentLine.append('path')
+      .classed('good-incident-line', true)
+      .attr('d', incidentLineGenerator(goodIncidentSeriesArray))
     
+    this.badIncidentLine.append('path')
+      .classed('bad-incident-line', true)
+      .attr('d', incidentLineGenerator(badIncidentSeriesArray))
+  }
 
-    // ENTER + UPDATE
-    goodBinding.merge(goodEntering)
-      .transition()
-      .call(incidentLineGenerator);
+  updateIncidentShading() {
+    const { goodIncidentSeries, badIncidentSeries, incidentShadingGenerator, series } = this.props;
+    
+    var goodIncidentShadingArray = [];
+    var badIncidentShadingArray = [];
+
+    series[0].results
+      .filter(entry => (entry.date.isSameOrAfter(goodIncidentSeries.start.clone().subtract(1, 'months')) && entry.date.isSameOrBefore(goodIncidentSeries.end.clone().add(1, 'months'))))
+      .map(entry => (
+        goodIncidentShadingArray.push(
+          { x: entry.date, 
+            y0: goodIncidentSeries.download_speed_mbps_median, 
+            y1: entry.download_speed_mbps_median
+          }
+        )
+      ));
+    
+    series[0].results
+      .filter(entry => (entry.date.isSameOrAfter(badIncidentSeries.start.clone().subtract(1, 'months')) && entry.date.isSameOrBefore(badIncidentSeries.end.clone().add(1, 'months'))))
+      .map(entry => (
+        badIncidentShadingArray.push(
+          { x: entry.date, 
+            y0: badIncidentSeries.download_speed_mbps_median, 
+            y1: entry.download_speed_mbps_median
+          }
+        )
+      ));
+    
+    this.goodIncidentShading.selectAll('*').remove();
+    this.badIncidentShading.selectAll('*').remove();
 
-    badBinding.merge(badEntering)
-      .transition()
-      .call(incidentLineGenerator);
+    this.goodIncidentShading.append('path')
+      .attr('d', incidentShadingGenerator(goodIncidentShadingArray));
 
-    // EXIT
-    goodBinding.exit().remove();
-    badBinding.exit().remove();
+    this.badIncidentShading.append('path')
+      .attr('d', incidentShadingGenerator(badIncidentShadingArray));
+    
+  
   }
 
   updateIncidentArrow() {
-    const { goodIncidentSeries, badIncidentSeries, incidentLineGenerator, xScale, yScale } = this.props;
+    const { goodIncidentSeries, badIncidentSeries, xScale, yScale } = this.props;
```

Comment:
Please add a top of method comment on how updateIncidentArrow calculates how to draw the arrow.